pkg/notifications/slack.go

```diff
@@ -0,0 +1,52 @@
+package notifications
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+func init() {
+	initers = append(initers, func(cfg map[string]string) Notifier {
+		if url, ok := cfg["slack_url"]; ok {
+			notifier := &slackNotifier{
+				URL: url,
+			}
+			return notifier
+		}
+		return nil
+	})
+}
+
+// slackNotifier sends notifications to slack or mattermost
+type slackNotifier struct {
+	URL      string
+	messages []string
+}
+
+func (s *slackNotifier) Notify(domain, provider, msg string, err error, preview bool) {
+	var content string
+	if preview {
+		content = fmt.Sprintf(`**Preview: %s[%s] -** %s`, domain, provider, msg)
+	} else if err != nil {
+		content = fmt.Sprintf(`**ERROR running correction on %s[%s] -** (%s) Error: %s`, domain, provider, msg, err)
+	} else {
+		content = fmt.Sprintf(`Successfully ran correction for **%s[%s]** - %s`, domain, provider, msg)
+	}
+	s.messages = append(s.messages, content)
+}
+
+func (s *slackNotifier) Done() {
```

Comment:
I noticed that you are collecting all the messages and then sending them at the end. That's fine.  However I was wondering if you tried sending them one at a time just to see how Slack displays them.

The reason I ask is that at Stackoverflow we once prevented an outage because we saw the notifications one-by-one.  The first few were deleting EVERYTHING.  Someone went into our CI/CD system and killed the job.  If we didn't have any notifications until the very end, nearly every DNS record related to stackoverflow.com would have been deleted.

Reply:
Good point, haven't thought about that case. Slack and also Mattermost should display them the same way as if i send them as a batch.

I will check that and change it to your suggestion.